qiskit_experiments/hamiltonian_tomography/cr_hamiltonian_experiment.py

```diff
@@ -0,0 +1,299 @@
+                The net CR pulse duration is determined by
+
+                .. math::
+
+                    \tau' = \tau - 2 {\rm (risefall)} \sigma + \sqrt{2\pi\sigma^2}
+
+                where :math:`\sigma` is the standard deviation of Gaussian edges and
+                `\tau` represents the total duration of the CR pulse.
+                If ``cr_gate_schedule`` is overridden without ``x_values``,
+                this will be identical to ``durations`` in units of second.
+
+        Raises:
+            QiskitError: When qubit number is not 2.
+            QiskitError: When ``cr_gate_schedule`` is specified without parameter named "duration".
+            QiskitError: When length of ``x_values`` and ``durations`` are different.
+        """
+        super().__init__(qubits)
+
+        if len(qubits) != 2:
+            raise QiskitError(
+                f"{self.__class__.__name__} is 2 qubit experiment. len({qubits}) != 2."
+            )
+
+        if cr_gate_schedule is not None and not cr_gate_schedule.get_parameters("duration"):
+            raise QiskitError(
+                "Target cross resonance schedule doesn't contain the parameter `duration`."
+            )
+
+        if x_values is not None and len(durations) != len(x_values):
+            raise QiskitError(
+                "Length of x values are not equivalent to durations; "
+                f"{len(self.experiment_options.durations)} != {len(x_values)}."
+            )
+
+        self._cr_gate_schedule = cr_gate_schedule
+        self._x_values = x_values
+
+        self.set_experiment_options(durations=durations)
+
+    @classmethod
+    def _default_experiment_options(cls) -> Options:
+        """Default option values used for CR Hamiltonian tomography."""
+        return Options(
+            amp=0.1,
+            sigma=64,
+            risefall=2,
+            durations=None,
+            alignment=16,
+            unit="dt",
+            dt=None,
+        )
+
+    def circuits(self, backend: Optional[Backend] = None) -> List[QuantumCircuit]:
+        """Create the circuits for the CR Hamiltonian experiment.
+
+        Args:
+            backend: A backend object.
+
+        Returns:
+            circuits: The circuits that will run the spectroscopy experiment.
+
+        Raises:
+            QiskitError: When `dt` information is not provided.
+        """
+        if self.experiment_options.dt is None:
+            try:
+                self.set_experiment_options(dt=backend.configuration().dt)
+            except AttributeError as ex:
+                raise QiskitError(
+                    "Duration is not set and the backend doesn't provide dt. "
+                    "Set time resolution of the system to proceed."
+                ) from ex
+
+        durations_dt = np.asarray(
+            [self._to_samples(dur) for dur in self.experiment_options.durations], dtype=int
+        )
+
+        if not self._cr_gate_schedule:
```

Comment:
Consider putting this in a method `_get_schedule` and do `cr_sched = self._get_schedule()` to keep the steps in `circuits` easy to follow as the circuits are the key part to understand the experiment.